### PR TITLE
Disable generic math due to C++/CLI incompatibilities

### PR DESCRIPTION
**Pull request by tannergooding**
This disables generic math as its not currently supported by C++/CLI and may cause `error C2253` (with a message similar to `'System::INumber::DivRem': pure specifier or abstract override specifier only allowed on virtual function`) for an imported type that implements an interface with static virtuals.

This is somewhat similar to what happens for C++/CLI when encountering `Span<T>` (or `ref structs` in general), `in`, and other "new" language or runtime features. However, since this directly impacts the "core" types (e.g. `int32`, `int64`, `intptr`, etc) it is much more prevalent and will likely be encountered by all existing C++/CLI codebases.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @danmoseley, @jeffhandley 

This is a draft PR so it can pass all normal CI validation and will be ready to merge before the snap tomorrow if its determined to be necessary.

https://github.com/dotnet/wpf/pull/4823 has some more details

**Review by stephentoub — APPROVED** [review]
😦

**Review by jeffhandley — APPROVED** [review]
1. Thanks for getting this ready in case we need it.
2. I'm actually excited to get to test our ability to turn this off. 😄
3. What is the best way to verify that all traces are removed?

**Comment by jkotas** [comment]
Have we looked for other potential places where the generic may cause problems? For example, debuggers and profilers?

**Comment by tannergooding** [comment]
> For example, debuggers and profilers?

For debugging at least, I didn't encounter any issues when implementing the feature (either product or test code). I didn't attempt any profiling.

I would suspect that, as with any new IL sequence (including things like `Default Interface Methods`) there may be external tooling updates required for the new feature.

**Comment by tannergooding** [comment]
> I'm actually excited to get to test our ability to turn this off. 😄
> What is the best way to verify that all traces are removed?

I had already done some testing locally before the feature was ever merged. Namely just using a version of Roslyn that didn't `support static abstracts in interfaces` and ensuring everything compiled.
I don't think there is a good way to test this otherwise short of manually examining the IL to ensure it looks as expected.

**Comment by tannergooding** [comment]
Marking as auto-merge. Due to this impacting essentially any C++/CLI compilation this has to be merged so that C++/CLI is usable from .NET 6.

We will be working with the C++/CLI team to get an appropriate fix in place.

**Comment by danmoseley** [comment]
We maybe should have a test hello world C++/CLI project in dotnet/runtime tests.

**Comment by tannergooding** [comment]
Do we currently have the infrastructure in place to build C++/CLI projects?

I imagine that we'll need quite a bit of special infrastructure/msbuild logic for a vcxproj to reference the locally built System.Runtime dependencies and account for other target/props differences.

@ViktorHofer?

**Comment by dreddy-work** [comment]
@tannergooding, Is anything blocking execution of tests? tests are taking forever and i see some tests failed.

**Comment by jkotas** [comment]
> We maybe should have a test hello world C++/CLI project in dotnet/runtime tests.

Does hello world C++/CLI hits this problem?

We have number of C++/CLI tests under https://github.com/dotnet/runtime/blob/main/src/tests/Interop/IJW/ . Is the problem that these tests are not getting compiled against current references?

**Comment by tannergooding** [comment]
> Does hello world C++/CLI hit this problem?

At the very least, it appears as though using any of the non-primitives that implement generic-math (including `IntPtr`) causes the problem. Its unclear under what conditions using the language primitives (such as `Int32`) also causes issues.

I'm working on getting a nightly SDK installed so I can do more analysis locally.

**Comment by tannergooding** [comment]
> Is anything blocking execution of tests? tests are taking forever and i see some tests failed.

The staging leg can take some time to complete. The last iteration timed out and so it was requeued and has been running for ~50m

**Comment by tannergooding** [comment]
@jkotas, use of any non-primitive type that implements an interface containing a `static abstract` method causes a failure.
Likewise, attempting to use any member on a primitive type (say `Int32.ToString`) will cause the same set of failures.
Attempting to just use a primitive type (e.g. `int main(array<System::String^>^ args)`) does not result in a failure.

The "simplest" repro for a failure is to install the latest .NET SDK nightly from dotnet/installer and then attempt to build the following in C++/CLI:
```cpp
using namespace System;

int main(array<System::String^>^ args)
{
  return Int32::Parse(args[0]);
}
```

Which results in the below. Ultimately it looks like the C++/CLI compiler attempts to validate every member of the type matches its expectations, even if the APIs in question are never used or consumed.
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2803	'operator &' must have at least one formal parameter of class type	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2803	'operator ^' must have at least one formal parameter of class type	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2803	'operator |' must have at least one formal parameter of class type	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2803	'operator ~' must have at least one formal parameter of class type	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IAdditionOperators::operator +': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IAdditionOperators<TSelf,TSelf,TSelf>::operator +': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IAdditiveIdentity::AdditiveIdentity::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IAdditiveIdentity<TSelf,TSelf>::AdditiveIdentity::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryInteger::LeadingZeroCount': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryInteger::PopCount': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryInteger::RotateLeft': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryInteger::RotateRight': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryInteger::TrailingZeroCount': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryNumber::IsPow2': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryNumber::Log2': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryNumber<TSelf>::IsPow2': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBinaryNumber<TSelf>::Log2': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators::operator &': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators::operator ^': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators::operator |': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators::operator ~': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<int,int,int>::operator &': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<int,int,int>::operator ^': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<int,int,int>::operator |': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<int,int,int>::operator ~': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<TSelf,TSelf,TSelf>::operator &': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<TSelf,TSelf,TSelf>::operator ^': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<TSelf,TSelf,TSelf>::operator |': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IBitwiseOperators<TSelf,TSelf,TSelf>::operator ~': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators::operator <': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators::operator <=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators::operator >': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators::operator >=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators<TSelf,TSelf>::operator <': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators<TSelf,TSelf>::operator <=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators<TSelf,TSelf>::operator >': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IComparisonOperators<TSelf,TSelf>::operator >=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IDecrementOperators::operator --': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IDecrementOperators<TSelf>::operator --': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IDivisionOperators::operator /': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IDivisionOperators<TSelf,TSelf,TSelf>::operator /': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators::operator !=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators::operator ==': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators<TSelf,TOther>::operator !=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators<TSelf,TOther>::operator ==': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators<TSelf,TSelf>::operator !=': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IEqualityOperators<TSelf,TSelf>::operator ==': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IIncrementOperators::operator ++': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IIncrementOperators<TSelf>::operator ++': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IModulusOperators::operator %': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IModulusOperators<TSelf,TSelf,TSelf>::operator %': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IMultiplicativeIdentity::MultiplicativeIdentity::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IMultiplicativeIdentity<TSelf,TSelf>::MultiplicativeIdentity::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IMultiplyOperators::operator *': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IMultiplyOperators<TSelf,TSelf,TSelf>::operator *': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Abs': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Clamp': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Create': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::CreateSaturating': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::CreateTruncating': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::DivRem': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Max': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Min': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::One::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Sign': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::TryCreate': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber::Zero::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Abs': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Clamp': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Create': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::CreateSaturating': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::CreateTruncating': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::DivRem': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Max': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Min': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::One::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Sign': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::TryCreate': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::INumber<TSelf>::Zero::get': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IParseable::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IParseable::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IParseable<TSelf>::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IParseable<TSelf>::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IShiftOperators::operator <<': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IShiftOperators::operator >>': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IShiftOperators<TSelf,TSelf>::operator <<': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IShiftOperators<TSelf,TSelf>::operator >>': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISpanParseable::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISpanParseable::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISpanParseable<TSelf>::Parse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISpanParseable<TSelf>::TryParse': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISubtractionOperators::operator -': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::ISubtractionOperators<TSelf,TSelf,TSelf>::operator -': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IUnaryNegationOperators::operator -': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IUnaryNegationOperators<TSelf,TSelf>::operator -': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IUnaryPlusOperators::operator +': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C2253	'System::IUnaryPlusOperators<TSelf,TSelf>::operator +': pure specifier or abstract override specifier only allowed on virtual function	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
Error	C1003	error count exceeds 100; stopping compilation	HelloWorldCppCLI	C:\Users\tagoo\Source\repos\HelloWorldCppCLI\HelloWorldCppCLI\main.cpp	5	
```

**Comment by RussKie** [comment]
> Have we looked for other potential places where the generic may cause problems? For example, debuggers and profilers?

You can add "compiling WPF" to the test suit whenever C++ gets changed.

**Comment by jkoritzinsky** [comment]
@jkotas the IJW tests build as part of the native test build, so they don't build against the live ref pack. We'd have to do significant build refactoring to enable them to build against the live ref pack in CI. Local builds are easier, so we could enable it for local builds as an opt-in option since it's only needed for cross-cutting features that happen once a release.

**Pull request merged**
